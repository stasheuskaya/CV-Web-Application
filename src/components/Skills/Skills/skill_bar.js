export default function SkillBar(props) {
    return (
        <div className="skillBar" style={{width: `${props.range}%`}}>
            {props.name}
        </div>
    )
}