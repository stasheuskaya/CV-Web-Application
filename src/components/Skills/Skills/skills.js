import '../skills.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPenToSquare } from '@fortawesome/free-solid-svg-icons';
import AddSkillForm from "../AddSkillForm/add_skill_form";
import { useSelector, useDispatch } from "react-redux";
import { toggleOpen } from '../../../store/actions/skills_actions';
import SkillBar from './skill_bar';

export default function Skills () {
    const showSkillForm = useSelector(state => state.skills.isOpen);
    const skills = useSelector(state => state.skills.items);
    const dispatch = useDispatch();

    return (
        <div className="skills">
            <button
                className='skillsEditButton'
                title="Open edit"
                icon={<FontAwesomeIcon icon={faPenToSquare} />}
                onClick={() => dispatch(toggleOpen())}>
                {<FontAwesomeIcon icon={faPenToSquare} />} Open edit
            </button>
            {showSkillForm && (<AddSkillForm />)}
            {skills.map(item => 
                <SkillBar 
                    name={item.name}
                    range={item.range}
            />)}
            <table className="skillTable">
                <tbody className="skillsLine">
                    <tr>
                        <th width="25%"></th>
                        <th></th>
                        <th width="25%"></th>
                    </tr>
                </tbody>
            </table>
            <div className="skillsLineText">
                <h1 className="beginnerSkill">
                    Beginner 
                </h1>
                <h1 className="masterSkill">
                    Master 
                </h1>
            </div>
            <div className="skillsLineText center">
                <h1 className="proficientSkill">
                    Proficient 
                </h1>
                <h1 className="expertSkill">
                    Expert
                </h1>
            </div>
        </div>
    )
}