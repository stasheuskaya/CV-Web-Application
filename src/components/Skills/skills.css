.inputErrorMessage {
    color: var(--red-error-color);
    margin-left: 20px;
}

.skills {
    position: relative;
  }
  
  .skillsEditButton {
    position: absolute;
    right: 40px;
    top: -75px;
    width: 140px;
    height: 40px;
  }
  
  .addSkillForm {
    width: 100%;
    height: 100%;
    border: 1px solid var(--green-color-highlight-shamrock);
    border-radius: 5px;
    padding-top: 15px;
    margin-bottom: 15px;
  }
    
  .skillName, .skillRange {
    display: flex;
    margin-left: 20px;
    padding-top: 30px;
    align-items: center;
  }
  
  .skillName input, .skillRange input {
    height: 30px;
    width:250px;
    font-size: 12px;
    margin-left: 20px;
  }
  
  .errBorder {
    border-color: var(--red-error-color);
  }
  
  .skillTable, .skillsLine, .skillsLine th {
    border: 1px solid var(--dark-grey-color-links-raven);
    border-top: none;
  }
  
  .skillTable {
    border-collapse: collapse;
    width: 100%;
    margin-top: 15px;
  }
  
  .skillsLine th {
    height: 10px;
  }
  
  .skillBar {
    height: 30px;
    background-color: var(--green-color-highlight-shamrock);
    color: var(--main-text-color-white);
    margin-top: 0px;
    margin-bottom: 5px;
    padding: 2px;
  }
  
  .skillsLineText {
    display: flex;
    padding-top: 10px;
    color: var(--dark-grey-color-links-raven);
    height: 30px;
  }
  
  .skillsLineText.center {
    position: relative;
    top: -40px;
  }
  
  .beginnerSkill, .proficientSkill, .expertSkill, .masterSkill {
    width: 50%;
  }
  
  .beginnerSkill {
    text-align: start;
  }
  
  .masterSkill {
    text-align: end;
  }
  
  .proficientSkill, .expertSkill {
    text-align: center;
  }
  