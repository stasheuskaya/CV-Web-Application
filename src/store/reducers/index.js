import { combineReducers } from "redux";
import { skillsReducer } from "./skills_reducer";
import { educationsReducer } from "./educations_reducer";

const allReducers = combineReducers({
    skills: skillsReducer,
    educations: educationsReducer
})

export default allReducers;