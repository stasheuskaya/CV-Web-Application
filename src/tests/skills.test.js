import renderer from "react-test-renderer";
import Skills from "../components/Skills/Skills/skills";
import { Provider } from 'react-redux';
import { store } from "../store/store"

describe('Skills Page', () => {
    it('renders content', () => {
		expect(renderer.create(
			<Provider store={store}>
                <Skills />
			</Provider>)).toMatchSnapshot();
	});
});