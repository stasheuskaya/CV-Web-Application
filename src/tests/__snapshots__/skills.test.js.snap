// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills Page renders content 1`] = `
<div
  className="skills"
>
  <button
    className="skillsEditButton"
    icon={
      <FontAwesomeIcon
        beat={false}
        beatFade={false}
        border={false}
        bounce={false}
        className=""
        fade={false}
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [
                "edit",
              ],
              "f044",
              "M490.3 40.4C512.2 62.27 512.2 97.73 490.3 119.6L460.3 149.7L362.3 51.72L392.4 21.66C414.3-.2135 449.7-.2135 471.6 21.66L490.3 40.4zM172.4 241.7L339.7 74.34L437.7 172.3L270.3 339.6C264.2 345.8 256.7 350.4 248.4 353.2L159.6 382.8C150.1 385.6 141.5 383.4 135 376.1C128.6 370.5 126.4 361 129.2 352.4L158.8 263.6C161.6 255.3 166.2 247.8 172.4 241.7V241.7zM192 63.1C209.7 63.1 224 78.33 224 95.1C224 113.7 209.7 127.1 192 127.1H96C78.33 127.1 64 142.3 64 159.1V416C64 433.7 78.33 448 96 448H352C369.7 448 384 433.7 384 416V319.1C384 302.3 398.3 287.1 416 287.1C433.7 287.1 448 302.3 448 319.1V416C448 469 405 512 352 512H96C42.98 512 0 469 0 416V159.1C0 106.1 42.98 63.1 96 63.1H192z",
            ],
            "iconName": "pen-to-square",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        maskId={null}
        pull={null}
        pulse={false}
        rotation={null}
        shake={false}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        titleId={null}
        transform={null}
      />
    }
    onClick={[Function]}
    title="Open edit"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-pen-to-square "
      data-icon="pen-to-square"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M490.3 40.4C512.2 62.27 512.2 97.73 490.3 119.6L460.3 149.7L362.3 51.72L392.4 21.66C414.3-.2135 449.7-.2135 471.6 21.66L490.3 40.4zM172.4 241.7L339.7 74.34L437.7 172.3L270.3 339.6C264.2 345.8 256.7 350.4 248.4 353.2L159.6 382.8C150.1 385.6 141.5 383.4 135 376.1C128.6 370.5 126.4 361 129.2 352.4L158.8 263.6C161.6 255.3 166.2 247.8 172.4 241.7V241.7zM192 63.1C209.7 63.1 224 78.33 224 95.1C224 113.7 209.7 127.1 192 127.1H96C78.33 127.1 64 142.3 64 159.1V416C64 433.7 78.33 448 96 448H352C369.7 448 384 433.7 384 416V319.1C384 302.3 398.3 287.1 416 287.1C433.7 287.1 448 302.3 448 319.1V416C448 469 405 512 352 512H96C42.98 512 0 469 0 416V159.1C0 106.1 42.98 63.1 96 63.1H192z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
     Open edit
  </button>
  <table
    className="skillTable"
  >
    <tbody
      className="skillsLine"
    >
      <tr>
        <th
          width="25%"
        />
        <th />
        <th
          width="25%"
        />
      </tr>
    </tbody>
  </table>
  <div
    className="skillsLineText"
  >
    <h1
      className="beginnerSkill"
    >
      Beginner
    </h1>
    <h1
      className="masterSkill"
    >
      Master
    </h1>
  </div>
  <div
    className="skillsLineText center"
  >
    <h1
      className="proficientSkill"
    >
      Proficient
    </h1>
    <h1
      className="expertSkill"
    >
      Expert
    </h1>
  </div>
</div>
`;
